### PR TITLE
STP-3303: Add release notes for SAT 2.4 to integration

### PR DESCRIPTION
**Pull request by jld-dehn-hpe**
## Summary and Scope

Added release notes documenting the changes in SAT 2.4 from the previous version SAT 2.3.

## Issues and Related PRs

* Resolves [STP-3303](https://jira-pro.its.hpecorp.net:8443/browse/STP-3303)
* Change will also be needed in `release/2.4`

## Testing

Spell check and lint check

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: docs/release_notes.md [hunk 1]
```diff
@@ -1,5 +1,175 @@
 # SAT Release Notes
 
+## Summary of Changes in SAT 2.4
+
+The 2.4.13 version of the SAT product includes:
+
+- Version 3.19.2 of the `sat` python package and CLI.
```
Comment (ekamin-hpe) [inline docs/release_notes.md:7]:
```suggestion
- Version 3.19.3 of the `sat` python package and CLI.
```

File: docs/release_notes.md [hunk 2]
```diff
@@ -1,5 +1,175 @@
...
+- Version 2.0.0-1 of the `sat-podman` wrapper script.
+- Version 1.5.4 of the `sat-install-utility` container image.
```
Comment (ekamin-hpe) [inline docs/release_notes.md:9]:
```suggestion
- Version 1.5.5 of the `sat-install-utility` container image.
```

File: docs/release_notes.md [hunk 3]
```diff
@@ -1,5 +1,175 @@
...
+- Version 3.3.0 of the `cfs-config-util` container image.
```
Comment (ekamin-hpe) [inline docs/release_notes.md:10]:
```suggestion
- Version 3.3.1 of the `cfs-config-util` container image.
```

File: docs/release_notes.md [hunk 4]
```diff
@@ -1,5 +1,175 @@
...
+This is the first release of SAT built from open source code repositories.
+As a result, build infrastructure was changed to use an external Jenkins instance,
+and artifacts are now published to an external Artifactory instance. These
+changes should not impact the functionality of the SAT product in any way.
+
+### Security
+
+#### CVE Mitigation
+
+The `paramiko` Python package version was updated from 2.9.2 to 2.10.1 to
```
Comment (ekamin-hpe) [inline docs/release_notes.md:144]: Another CVE item could be added here - 

> The `oauthlib` Python package version was updated from 3.2.0 to 3.2.1 to mitigate CVE-2022-36087.

https://github.com/Cray-HPE/sat/blob/release/3.19/CHANGELOG.md#security

**Review by ekamin-hpe — CHANGES_REQUESTED** [review]
A few comments, mostly related to last-minute bug fixes incrementing versions of things.

As for the spell check failures, I think it would be fine to add "traceback", "Artifactory" and CVE ID(s) into the `.spelling` file to define them as exceptions for the spell check.

With the remaining spell check failures, I think we are seeing the issue I described in this conversation: https://github.com/Cray-HPE/docs-sat/pull/34#discussion_r988327357

I think there is some kind of bug in the spell check when it comes to handling links that contain line breaks. For the most part, I think if you just put the entire link on one line that will address these failures, though in the example of lines 33-34 I think that would make the line too long.

There could be a creative way to resolve this (e.g. shortening the link text), but if not I would say it's fine to add e.g. "decoupling-of-cray-sat-container-image-and-cray-sat-podman-package" to the spelling exceptions file as well, ideally with a comment saying this is due to a bug with the spell checker. Even though the spell checker is being a bit too strict here I would prefer to try and keep it passing.

**Review by jack-stanek-hpe — APPROVED** [review]
My comments are the same as Eli's. Other than those, I think this looks good!

**Comment by jld-dehn-hpe** [comment]
@ekamin-hpe, I added your changes and got all the checks to pass. Let me know if there's any other changes you want before approving. Thanks!

**Review by ekamin-hpe — APPROVED** [review]
> @ekamin-hpe, I added your changes and got all the checks to pass. Let me know if there's any other changes you want before approving. Thanks!

Looks good, thanks for taking the time to address both the feedback and the failing checks. Glad it worked!!

**Pull request merged**